src/librustdoc/html/static/js/search.js

```diff
@@ -404,17 +405,17 @@ window.initSearch = function(rawSearchIndex) {
                 // a levenshtein distance value that isn't *this* good so it goes
                 // into the search results but not too high.
                 lev_distance = Math.ceil((checkGenerics(obj, val) + lev_distance) / 2);
-            } else if (obj.length > GENERICS_DATA && obj[GENERICS_DATA].length > 0) {
+            }
+            if (obj.length > GENERICS_DATA && obj[GENERICS_DATA].length > 0) {
                 // We can check if the type we're looking for is inside the generics!
                 var olength = obj[GENERICS_DATA].length;
                 for (x = 0; x < olength; ++x) {
-                    lev_distance = Math.min(levenshtein(obj[GENERICS_DATA][x], val.name),
-                                            lev_distance);
+                    tmp_lev = Math.min(levenshtein(obj[GENERICS_DATA][x][NAME], val.name), tmp_lev);
```

Comment:
Wouldn't we need to recurse at this point to check whether the generics of the generic match the searched generics?

Reply:
Absolutely, it's just that I wanted to keep the current (broken) behaviour for the moment. I plan to add it into the rework of the search engine. But if you prefer that I make this change now, I can as well.

Reply:
Done!